pyiceberg/table/update/validate.py

```diff
@@ -69,3 +75,74 @@ def validation_history(
         raise ValidationException("No matching snapshot found.")
 
     return manifests_files, snapshots
+
+
+def deleted_data_files(
+    table: Table,
+    starting_snapshot: Snapshot,
+    data_filter: Optional[BooleanExpression],
+    parent_snapshot: Optional[Snapshot],
+    partition_set: Optional[set[Record]],
+) -> Iterator[ManifestEntry]:
+    """Find deleted data files matching a filter since a starting snapshot.
+
+    Args:
+        table: Table to validate
+        starting_snapshot: Snapshot current at the start of the operation
+        data_filter: Expression used to find deleted data files
+        partition_set: a set of partitions to find deleted data files
+        parent_snapshot: Ending snapshot on the branch being validated
+
+    Returns:
+        List of deleted data files matching the filter
+    """
+    # if there is no current table state, no files have been deleted
+    if parent_snapshot is None:
+        return
+
+    manifests, snapshot_ids = validation_history(
+        table,
+        starting_snapshot,
+        parent_snapshot,
```

Comment:
This looks out of order to, because `validation_history` has `to_snapshot` as the second argument, and `from_snapshot` as the third argument.
```suggestion
        starting_snapshot,
        parent_snapshot,
```

Do you think it would be better to update `validation_history` function to use the following function signature instead? I think it's a lot more expected to have `from_snapshot` then `to_snapshot`

```
def validation_history(
    table: Table,
    from_snapshot: Snapshot,
    to_snapshot: Snapshot,
    matching_operations: set[Operation],
    manifest_content_filter: ManifestContent,
)
```

Reply:
Could you clarify a little more? Are you saying we should move to the terms `starting_snapshot` (which is `from_snapshot`) and `parent_snapshot` (`to_snapshot`)